### PR TITLE
Misc release fixes

### PR DESCRIPTION
**Pull request by albi3ro**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      test directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `doc/releases/changelog-dev.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] The PennyLane source code conforms to
      [PEP8 standards](https://www.python.org/dev/peps/pep-0008/).
      We check all of our code against [Pylint](https://www.pylint.org/).
      To lint modified files, simply `pip install pylint`, and then
      run `pylint pennylane/path/to/file.py`.

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: doc/releases/changelog-0.45.0.md [hunk 1]
```diff
@@ -936,7 +936,9 @@
 * The ``BoundTransform.transform`` property has been deprecated. Use ``BoundTransform.tape_transform`` instead.
   [(#8985)](https://github.com/PennyLaneAI/pennylane/pull/8985)
 
-* :func:`~pennylane.tape.qscript.expand` and the related functions :func:`~pennylane.tape.expand_tape`, :func:`~pennylane.tape.expand_tape_state_prep`, and :func:`~pennylane.tape.create_expand_trainable_multipar`
+* :meth`~QuantumScript.expand`, :func:`~pennylane.tape.qscript.expand` and the related functions
+  :func:`~pennylane.tape.expand_tape`, :func:`~pennylane.tape.expand_tape_state_prep`, and
+  :func:`~pennylane.tape.create_expand_trainable_multipar`
   have been deprecated and will be removed in v0.46. Instead, please use the :func:`qp.transforms.decompose <.transforms.decompose>`
   function for decomposing circuits.
   [(#8943)](https://github.com/PennyLaneAI/pennylane/pull/8943)
```
Comment (isaacdevlugt) [inline doc/releases/changelog-0.45.0.md:951]:
```suggestion
  [(#8943)](https://github.com/PennyLaneAI/pennylane/pull/8943)
  [(#9438)](https://github.com/PennyLaneAI/pennylane/pull/9438)
```

File: doc/introduction/importing_workflows.rst [hunk 2]
```diff
@@ -107,7 +108,65 @@ It can also be visualized using PennyLane's :func:`~pennylane.draw` utility:
 OpenQASM
 ~~~~~~~~
 
-An equivalent quantum circuit can be expressed in OpenQASM 2.0 as follows:
+:func:~pennylane.io.from_qasm3` supports the OpenQASM 3.0 spec, including control flow,
+mid circuit measurements, subroutines, classical computation, and more.
```
Comment (comp-phys-marc) [inline doc/introduction/importing_workflows.rst]: Could provide a little more info? i.e. 

The standard library gates, qubit registers, built-in mathematical functions and constants, subroutines, variables, control flow, measurements, inputs, outputs, custom gates and ``end`` statements are supported.

File: doc/introduction/importing_workflows.rst [hunk 3]
```diff
@@ -107,7 +108,65 @@ It can also be visualized using PennyLane's :func:`~pennylane.draw` utility:
...
+            {'q0': 0, 'q1': 1, 'q2': 2}
+        )()
+        return qp.expval(qp.Z(0))
+
+Inspecting the circuit, we can see that the operations and measurements have been correctly interpreted.
+
+>>> print(qp.draw(my_circuit)())
+0: ──RY(0.10)──X²────────────┤  <Z>
+1: ──RX(0.20)───────RZ(1.00)─┤
+2: ──H─────────┤↗├──║────────┤
```
Comment (comp-phys-marc) [inline doc/introduction/importing_workflows.rst:165]: Nice :+1:

File: doc/introduction/inspecting_circuits.rst [hunk 4]
```diff
@@ -53,26 +53,27 @@ For example:
 We can now use the :func:`~pennylane.specs` transform to generate a function that returns
 details and resource information:
 
->>> x = np.array([0.05, 0.1, 0.2, 0.3], requires_grad=True)
->>> y = np.array(0.4, requires_grad=False)
+>>> x = qp.numpy.array([0.05, 0.1, 0.2, 0.3], requires_grad=True)
+>>> y = qp.numpy.array(0.4, requires_grad=False)
```
Comment (comp-phys-marc) [inline doc/introduction/inspecting_circuits.rst:57]: I thought we were moving away from using `qp.numpy`?

File: doc/introduction/inspecting_circuits.rst [hunk 5]
```diff
@@ -53,26 +53,27 @@ For example:
...
+Total gates: 4
+Gate counts:
+- RX: 1
+- Toffoli: 1
+- CRY: 1
+- Rot: 1
+Measurements:
+- expval(PauliZ): 1
+- expval(PauliX): 1
+Depth: 4
```
Comment (comp-phys-marc) [inline doc/introduction/inspecting_circuits.rst:75]: :+1:

File: doc/introduction/templates.rst [hunk 6]
```diff
@@ -268,7 +268,7 @@ Other subroutines
 Other useful templates which do not belong to the previous categories can be found here.
 
 .. gallery-item::
-    :description: :doc:`Instantaneous Quantum Polynomial Circuit <../code/api/pennylane.IQP>`
+    :description: :doc:`Instantaneous Quantum Polynomial Circuit (IQP) <../code/api/pennylane.IQP>`
```
Comment (comp-phys-marc) [inline doc/introduction/templates.rst:271]: thanks

File: pennylane/transforms/optimization/merge_amplitude_embedding.py [hunk 7]
```diff
@@ -374,39 +374,13 @@ def qfunc():
...
-
-        Using the transformation we can join the different amplitude embedding into a single one:
-
         >>> optimized_qfunc = qp.transforms.merge_amplitude_embedding(qfunc)
         >>> optimized_qnode = qp.QNode(optimized_qfunc, dev)
         >>> print(qp.draw(optimized_qnode)())
         0: ─╭●───┤  State
         1: ─╰X───┤  State
         2: ─╭|Ψ⟩─┤  State
         3: ─╰|Ψ⟩─┤  State
```
Comment (comp-phys-marc) [inline pennylane/transforms/optimization/merge_amplitude_embedding.py:387]: Wasn't @isaacdevlugt handling this change in his PR?

File: doc/introduction/importing_workflows.rst (hunk 2 above)
Comment (albi3ro) [reply doc/introduction/importing_workflows.rst]: The idea is to provide just enough information to get people to go point of the function itself.

File: doc/introduction/inspecting_circuits.rst (hunk 4 above)
Comment (albi3ro) [reply doc/introduction/inspecting_circuits.rst:57]: The point was to just get the example to follow our current import pattern, rather than changing anything in particular about the example.

But given we are no longer reporting the number of gradient executions, we can probably switch to vanilla numpy.

File: doc/introduction/importing_workflows.rst (hunk 2 above)
Comment (comp-phys-marc) [reply doc/introduction/importing_workflows.rst]: sure!

File: doc/introduction/inspecting_circuits.rst (hunk 4 above)
Comment (comp-phys-marc) [reply doc/introduction/inspecting_circuits.rst:57]: 👍

File: pennylane/transforms/optimization/merge_amplitude_embedding.py (hunk 7 above)
Comment (albi3ro) [reply pennylane/transforms/optimization/merge_amplitude_embedding.py:387]: I believe he pushed this change actually? 🤷

Comment (comp-phys-marc) [reply pennylane/transforms/optimization/merge_amplitude_embedding.py:387]: Oh I see. Well the change is in two different PRs at this point.

Comment (comp-phys-marc) [reply pennylane/transforms/optimization/merge_amplitude_embedding.py:387]: Maybe we can just revert it in one of them?

Comment (comp-phys-marc) [reply pennylane/transforms/optimization/merge_amplitude_embedding.py:387]: I'm ready to approve once this is resolved.

File: doc/releases/changelog-0.45.0.md [hunk 8]
```diff
@@ -940,10 +940,13 @@
 * The ``BoundTransform.transform`` property has been deprecated. Use ``BoundTransform.tape_transform`` instead.
   [(#8985)](https://github.com/PennyLaneAI/pennylane/pull/8985)
 
-* :func:`~pennylane.tape.qscript.expand` and the related functions :func:`~pennylane.tape.expand_tape`, :func:`~pennylane.tape.expand_tape_state_prep`, and :func:`~pennylane.tape.create_expand_trainable_multipar`
+* :meth`~QuantumScript.expand`, :func:`~pennylane.tape.qscript.expand` and the related functions
```
Comment (JerryChen97) [inline doc/releases/changelog-0.45.0.md]:
```suggestion
* :meth:`~QuantumScript.expand`, :func:`~pennylane.tape.qscript.expand` and the related functions
```

File: pennylane/transforms/optimization/merge_amplitude_embedding.py (hunk 7 above)
Comment (isaacdevlugt) [reply pennylane/transforms/optimization/merge_amplitude_embedding.py:387]: hey guys! Sorry I did indeed make this change in this PR: https://github.com/PennyLaneAI/pennylane/pull/9441/

Apologies for the duplicity 😵‍💫

Comment (comp-phys-marc) [reply pennylane/transforms/optimization/merge_amplitude_embedding.py:387]: no worries, I just reverted it for you https://github.com/PennyLaneAI/pennylane/pull/9438/commits/10c31dedba80e4c4c8d13412316800cfde1dae40

**Review by comp-phys-marc — APPROVED** [review]
Nice assortment of fixes

File: doc/releases/changelog-0.45.0.md [hunk 9]
```diff
@@ -941,10 +941,13 @@
 * The ``BoundTransform.transform`` property has been deprecated. Use ``BoundTransform.tape_transform`` instead.
   [(#8985)](https://github.com/PennyLaneAI/pennylane/pull/8985)
 
-* :func:`~pennylane.tape.qscript.expand` and the related functions :func:`~pennylane.tape.expand_tape`, :func:`~pennylane.tape.expand_tape_state_prep`, and :func:`~pennylane.tape.create_expand_trainable_multipar`
+* :meth:`~QuantumScript.expand`, :func:`~pennylane.tape.qscript.expand` and the related functions
```
Comment (JerryChen97) [inline doc/releases/changelog-0.45.0.md]: cross-ref rendering is fixed but they still look like two `expand` exactly identical:

Image

Comment (JerryChen97) [reply doc/releases/changelog-0.45.0.md]: Would the named ref e.g. ```:func:`qp.transforms.decompose <.transforms.decompose>` ``` work in changelog as well?

File: doc/introduction/importing_workflows.rst [hunk 10]
```diff
@@ -107,7 +108,65 @@ It can also be visualized using PennyLane's :func:`~pennylane.draw` utility:
 OpenQASM
 ~~~~~~~~
 
-An equivalent quantum circuit can be expressed in OpenQASM 2.0 as follows:
+:func:~pennylane.io.from_qasm3` supports the OpenQASM 3.0 spec, including control flow,
```
Comment (JerryChen97) [inline doc/introduction/importing_workflows.rst]:
```suggestion
:func:`~pennylane.io.from_qasm3` supports the OpenQASM 3.0 spec, including control flow,
```

**Review by JerryChen97 — APPROVED** [review]

File: doc/releases/changelog-0.45.0.md (hunk 9 above)
Comment (JerryChen97) [inline doc/releases/changelog-0.45.0.md]:
```suggestion
* :meth:`QuantumScript.expand <pennylane.tape.QuantumScript.expand>`, :func:`qp.tape.qscript.expand <pennylane.tape.qscript.expand>` and the related functions
```

**Review by isaacdevlugt — APPROVED** [review]
